src/pretix/static/pretixcontrol/js/ui/main.js

```diff
@@ -314,6 +314,31 @@ var form_handlers = function (el) {
         dependency.closest('.form-group').find('input[name=' + dependency.attr("name") + ']').on("dp.change", update);
     });
 
+    $("input[name$=vat_id]").each(function () {
+        var dependent = $(this),
+            dependency_country = $(this).closest(".panel-body, form").find('select[name$=country]'),
+            dependency_id_is_business_1 = $(this).closest(".panel-body, form").find('input[id$=id_is_business_1]'),
+            update = function (ev) {
+                if (dependency_id_is_business_1.length && !dependency_id_is_business_1.prop("checked")) {
+                    dependent.closest(".form-group").hide();
+                } else {
+                    dependent.prop("disabled", true);
+                    dependency_country.closest(".form-group").find("label").prepend("<span class='fa fa-cog fa-spin'></span> ");
+                    $.getJSON('/js_helpers/is_country_in_eu/?country=' + dependency_country.val(), function (data) {
+                        if (data.data) {
+                            dependent.closest(".form-group").show();
+                        } else {
+                            dependent.closest(".form-group").hide();
+                        }
+                        dependent.prop("disabled", false);
+                        dependency_country.closest(".form-group").find("label .fa-spin").remove();
+                    });
+                }
+            };
+        dependency_country.on("change", update);
+        dependency_id_is_business_1.on("change", update);
```

Comment:
We probably also need one ``update();`` call to evaluate it after page load (i.e. if something is pre-set)

Reply:
```suggestion
        dependency_id_is_business_1.on("change", update);
        $(window).on("load", update);
```

Reply:
No, ``load`` is pretty late. ``load`` only happens after *all* resources on the page have been loaded, which can take a while. You can just call ``update();`` – we're in a ``$(function() { … })`` block and jQuery ensures the DOM is ready.

Reply:
Thanks for the explanation, changed it.